### PR TITLE
ktm-9: Icon component

### LINKED ISSUE #9: Icon component

### PR DESCRIPTION
**Pull request by HIMDanny**
Closes #9

### CODE REVIEW

**Review by maxmotsyk — APPROVED** [review]

**Review by kuslosolodka — COMMENTED** [review]
We can import svg icons as svg and create some object like iconToSvg, in which we imported svg icon as react component and adding our icon with name, and use it in icon component, in which we are assign icon type as property to this iconToSvg object. This way is give us opportunity to not create separate component for every icon. Something like that:
```
import { ReactComponent as Icon } from './icon.svg'

const iconToSvg: Record<
  IconType,
  React.FC<React.SVGProps<SVGSVGElement>>
> = {'icon': Icon}

const Icon: React.FC<Properties> = ({ iconName }: Properties) => {
  const SvgIcon = iconToSvg[iconName];

  return <SvgIcon  />;
};

```

File: client/src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+type IconType =
+  | 'board'
+  | 'cancel'
+  | 'check'
+  | 'chevron'
+  | 'eye'
+  | 'eye-slash'
+  | 'menu'
+  | 'moon'
+  | 'sun';
```
Comment (kuslosolodka) [inline client/src/components/Icon/Icon.tsx]: separate this to types, pls

File: client/src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+interface IconProperties {
+  icon: IconType;
+}
+
+// eslint-disable-next-line react/prop-types
```
Comment (kuslosolodka) [inline client/src/components/Icon/Icon.tsx]: add this rule to eslint file with off value

**Comment by HIMDanny** [comment]
> 

We should install an additional plugin to Vite then because by default Vite doesn't have svg as React component support

**Comment by HIMDanny** [comment]
As an alternative, I can change the switch approach to an object map with already created react svg components

File: client/src/components/Icon.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const icons: Record<IconType, React.FC<React.SVGProps<SVGSVGElement>>> = {
+  board: BoardIcon,
+  cancel: CancelIcon,
+  check: CheckIcon,
+  chevron: ChevronDownIcon,
+  eye: EyeIcon,
+  'eye-slash': EyeSlashIcon,
+  menu: MenuIcon,
+  moon: MoonIcon,
+  sun: SunIcon,
```
Comment (kuslosolodka) [inline client/src/components/Icon.tsx]: So we need strings or not?)

**Review by kuslosolodka — CHANGES_REQUESTED** [review]

File: client/src/components/Icon.tsx (hunk 3 above)
Comment (HIMDanny) [reply client/src/components/Icon.tsx]: What do you mean?)

**Review by kuslosolodka — APPROVED** [review]

**Review by maxmotsyk — APPROVED** [review]

**Pull request merged**